src/realm/obj.cpp

```diff
@@ -991,9 +991,7 @@ void out_mixed_xjson(std::ostream& out, const Mixed& val)
         }
         case type_Timestamp: {
             out << "{\"$date\": {\"$numberLong\": \"";
-            auto ts = val.get<Timestamp>();
-            int64_t timeMillis = ts.get_seconds() * 1000 + ts.get_nanoseconds() / 1000000;
-            out << timeMillis;
+            out << util::serializer::print_value(val.get<Timestamp>());
```

Comment:
This is not the correct line to change. The linked issue is complaining about `Obj::to_json()` which uses `Mixed::to_json()` under the hood.

Reply:
`Obj::to_json` calls this `void out_mixed(std::ostream& out, const Mixed& val, JSONOutputMode output_mode)` ... maybe you are referring to something different? Also, `Mixed::to_json()` does not seem to exist, am I wrong?

Reply:
Apologies, I was looking at code in next-major which has diverged.
`Obj::to_json` does call `out_mixed_xjson` but only in xjson mode. I'm sure that the mode in the linked issue is not xjson but rather `output_mode_json` because in the issue description the reported error is about a human readable timestamp in the format `-001-01-01 00:00:00` but the output of xjson mode is just a number since epoch. I think the place for this change is above, in `out_mixed_json()`.

Reply:
Ah yeah, we changed that for nested collections to allow JSON into Mixed basically. Eventually this will have to be ported in the next-major as well.